h2o-r/h2o-package/R/frame.R

```diff
@@ -1716,7 +1716,8 @@ h2o.tail <- function(x,n=6L,...) {
   if( n==0L ) head(x,n=0L)
   else {
     startidx <- max(1L, endidx - n + 1)
-    as.data.frame(.newExpr("rows",x,paste0("[",startidx-1,":",(endidx-startidx+1),"]")))
+    df = .newExpr("rows",x,paste0("[",startidx-1,":",(endidx-startidx+1),"]"))
+    head(.fetch.data(df,startidx-1),n)
```

Comment:
I'm just utilizing head again. If you return a frame, then Log.info() breaks in a unit test as it will try to paste with an H2OFrame, which won't work.